File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android/com/example/smoke/PosDefaultsWithOneInternalField.java
/*

 *
 */

package com.example.smoke;

import android.support.annotation.NonNull;

/**
 * <p>Test case 4
 * <p>Input:
 * <ul>
 * <li>The structure uses <code>PositionalDefaults</code> and should get such constructor.</li>
 * <li>This structure does not define other constructors / field constructors.</li>
 * <li>It has one public field with default value.</li>
 * <li>One of fields is internal.</li>
 * </ul>
 * <p>Expected outcome:
 * <ul>
 * <li>Generated positional defaults constructor is not public (because it would expose internal field).</li>
 * </ul>
 */
public final class PosDefaultsWithOneInternalField {
    @NonNull
    String internalField;
    public boolean publicField;
    public int initializedField;

    PosDefaultsWithOneInternalField(@NonNull final String internalField, final boolean publicField) {
        this.internalField = internalField;
        this.publicField = publicField;
        this.initializedField = 77;
    }

    public PosDefaultsWithOneInternalField(@NonNull final String internalField, final boolean publicField, final int initializedField) {
        this.internalField = internalField;
        this.publicField = publicField;
        this.initializedField = initializedField;
    }


}

File: gluecodium/src/test/resources/smoke/visibility_attribute/output/android-kotlin/com/example/smoke/PosDefaultsWithOneInternalField.kt
/*

 *
 */

@file:JvmName("PosDefaultsWithOneInternalFieldExtensions")


package com.example.smoke


/**
 * Test case 4
 *
 * Input:
 * - The structure uses `PositionalDefaults` and should get such constructor.
 * - This structure does not define other constructors / field constructors.
 * - It has one public field with default value.
 * - One of fields is internal.
 *
 * Expected outcome:
 * - Generated positional defaults constructor is not public (because it would expose internal field).
 */
class PosDefaultsWithOneInternalField {
    @JvmField @JvmSynthetic internal var internalField: String
    @JvmField var publicField: Boolean
    @JvmField var initializedField: Int



    @JvmOverloads
    constructor(internalField: String, publicField: Boolean, initializedField: Int = 77) {
        this.internalField = internalField
        this.publicField = publicField
        this.initializedField = initializedField
    }





}

